plugin/pkg/shoot/validator/admission.go

```diff
@@ -723,9 +738,24 @@ func validateMachineImagesConstraints(constraints []core.MachineImage, image, ol
 	return false, validValues
 }
 
-func getOldWorkerMachineImageOrDefault(workers []core.Worker, name string, defaultImage *core.ShootMachineImage) *core.ShootMachineImage {
-	if oldWorker := helper.FindWorkerByName(workers, name); oldWorker != nil && oldWorker.Machine.Image != nil {
-		return oldWorker.Machine.Image
+func getOldWorkerMachineImageOrDefault(oldWorkers []core.Worker, worker core.Worker, images []core.MachineImage) (*core.ShootMachineImage, error) {
```

Comment:
Can we name it `ensureMachineImage` or similar? Because now it also handles the `worker.Machine.Image != nil` cases which was previously done by the caller.

Reply:
done